src/components/Helpers/FilterTodos.tsx

```diff
@@ -0,0 +1,32 @@
+import { Todo } from '../../types/Todo';
+
+export function filterTodos(
+  todos: Todo[],
+  sortBy: string,
+  query: string,
+): Todo[] {
+  const searchQuery = query.trim().toLowerCase();
+
+  const filtered = todos.filter((todo) => {
+    switch (sortBy) {
+      case 'active':
+        return !todo.completed;
+
+      case 'completed':
+        return todo.completed;
+
+      case 'all':
+      default:
+        return todo;
+    }
+  });
+
+  if (query) {
+    return filtered
+      .filter((todo) => todo
+        .title.toLowerCase()
+        .includes(searchQuery));
+  }
+
+  return filtered;
```

Comment:
you can get rid of if block, if there is nothing in query, and you make sure there are no whitespaces, this function will work fine without it